doc/tutorials/cmerge.md

```diff
@@ -0,0 +1,239 @@
+
+```sh
+kdb set user/tests/base a
+#> Create a new key user/tests/base with string "a"
+kdb set user/tests/their a
+#> Create a new key user/tests/their with string "a"
+kdb set user/tests/our a
+#> Create a new key user/tests/our with string "a"
+kdb cmerge user/tests/our user/tests/their user/tests/base user/tests/result
+kdb get user/tests/result
+#> a
+```
+
+We change the our version for another example.
+
+```sh
+kdb set user/tests/our b
+#> Set string to "b"
+```
+
+Using a `result` path that is not empty gives an error.
+The option `--force` can be used to override. **Attention!** This deletes existing keys below `result`.
+
+```sh
+kdb cmerge user/tests/our user/tests/their user/tests/base user/tests/result
+# RET: 1
+# STDERR: ERROR: 1 keys exist in resultpath. Use -f to override the keys there.
+kdb cmerge -f user/tests/our user/tests/their user/tests/base user/tests/result
+kdb get user/tests/result
+#> b
+```
+
+We can use the same key for different versions.
+
+```sh
+kdb set user/tests/same a
+#> Create a new key user/tests/same with string "a"
+kdb cmerge -f user/tests/same user/tests/same user/tests/same user/tests/result
+kdb get user/tests/result
+#> a
+```
+
+### hosts
+
+Merging is also possible in more complicated situations.
+As a real-world example, we import three different (see the comment) versions of a hosts file. Please remove the trailing backslashes from the example code.
+
+```sh
+echo "127.0.0.1       localhost\
+127.0.1.1       computer\
+\
+# BASE The following lines are desirable for IPv6 capable hosts\
+::1     ip6-localhost ip6-loopback\
+fe00::0 ip6-localnet\
+ff00::0 ip6-mcastprefix\
+ff02::1 ip6-allnodes\
+ff02::2 ip6-allrouters" | kdb import user/tests/hosts/base hosts
+
+echo "127.0.0.1       localhost\
+127.0.1.1       computer\
+\
+# OUR The following lines are desirable for IPv6 capable hosts\
+::1     ip6-localhost ip6-loopback\
+fe00::0 ip6-localnet\
+ff00::0 ip6-mcastprefix\
+ff02::1 ip6-allnodes\
+ff02::2 ip6-allrouters" | kdb import user/tests/hosts/our hosts
+
+echo "127.0.0.1       localhost\
+127.0.1.1       computer\
+\
+# THEIR The following lines are desirable for IPv6 capable hosts\
+::2     ip6-localhost ip6-loopback\
+fe00::0 ip6-localnet\
+ff00::0 ip6-mcastprefix\
+ff02::1 ip6-allnodes\
+ff02::2 ip6-allrouters" | kdb import user/tests/hosts/their hosts
+
+kdb cmerge user/tests/hosts/our user/tests/hosts/their user/tests/hosts/base user/tests/hosts/result
+kdb get user/tests/hosts/result/ipv6/ip6-localhost
+#> ::2
+```
+
+### Metadata
+
+Metadata gets merged as well.
+We do not follow a complicated approach for this topic.
+When a key-value pair is chosen from the three versions to be present in the result it takes all its metadata with it.
+
+In case that the values of some keys are equal, the `our` version wins and consequently the metadata of the `our` version is used. The goal of this is to preserve a user's personal notes for a key.
```

Comment:
Do you mean meta-keys here? Do not use the term "notes" here, consistently use "comments". In general it is very important to always stick to the same terms.